includes/Core/Authentication/Setup_V2.php

```diff
@@ -54,15 +54,23 @@ public function handle_action_setup_start() {
 			wp_die( esc_html__( 'Site Kit is not configured to use the authentication proxy.', 'google-site-kit' ) );
 		}
 
-		if ( $this->credentials->has() ) {
-			$this->google_proxy->sync_site_fields( $this->credentials, 'sync' );
+		$required_scopes = $this->authentication->get_oauth_client()->get_required_scopes();
+		$this->google_proxy->with_scopes( $required_scopes );
+
+		$oauth_setup_redirect = $this->credentials->has()
+			? $this->google_proxy->sync_site_fields( $this->credentials, 'sync' )
+			: $this->google_proxy->register_site();
```

Comment:
Just pointing out here that, related to my comments above, we may need to provide `'sync'` here if we introduce a parameter for it - I think that would be more consistent.

Reply:
Updated.